.changeset/poor-turtles-remember.md

```diff
@@ -0,0 +1,5 @@
+---
+"@itwin/itwinui-css": patch
+---
+
+Remove unnecessary gap below `.iui-input-grid` when `data-iui-label-placement='inline'` and no secondary line (`iui-status-message`) is present.
```

Comment:
nit/optional: After re-reading this, I felt my suggestion could lead to two interpretations. So, I think we can make it clearer (optional)

```suggestion
Remove unnecessary gap below `.iui-input-grid` when `data-iui-label-placement='inline'` and when no secondary line (`iui-status-message`) is present.
```